### DISCUSSION TITLE
Occurs intermittent errors when logging in from Google SAML.

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#5006] [Plugins]
Hello

When logging in Kimai Google SAML, the following error occurs intermittently.

![스크린샷 2024-08-02 172739](https://github.com/user-attachments/assets/189039ec-e646-4609-be2e-774749a218fd)

This is the contents of the Google SAML config file below.
/kimai/config/packages/local.yaml

kimai:
    calendar:
        google:
            api_key: '  api_key  '
            sources:
                holidays:
                    id:  ' calendar  '
                    color: '#ccc'
                company:
                    id: ' company calendar'
                    color: '#cc0000'
    saml:
        provider: google
        activate: true
        title: Login with SAML
        mapping:
            - { saml: $Email, kimai: email }
        roles:
            resetOnLogin: false
            attribute: Groups
            mapping:
                - { saml: Admin, kimai: ROLE_SUPER_ADMIN }
                - { saml: Manager, kimai: ROLE_ADMIN }
                - { saml: Teamlead, kimai: ROLE_TEAMLEAD }
        connection:
            idp:
                entityId: 'https://accounts.google.com/o/saml2?idpid=C01bx2wqs'
                singleSignOnService:
                    url: 'https://accounts.google.com/o/saml2/idp?idpid=C01bx2wqs'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                #singleLogoutService:
                #    url: 'https://www.example.com/logout'
                #    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                x509cert: '-----BEGIN CERTIFICATE-----
                              .....
                           -----END CERTIFICATE-----'
            # Your Kimai: replace https://www.example.com with your base URL
            sp:
                entityId: 'https://www.example.com/auth/saml/metadata'
                assertionConsumerService:
                    url: 'https://www.example.com/auth/saml/acs'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST'
                singleLogoutService:
                    url: 'https://www.example.com/auth/saml/logout'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                #privateKey: ''
             # only set baseurl, if auto-detection doesn't work
            #baseurl: ''
            strict: true
            debug: true
            security:
                nameIdEncrypted: false
                authnRequestsSigned: false
                logoutRequestSigned: false
                logoutResponseSigned: false
                wantMessagesSigned: false
                wantAssertionsSigned: false
                wantNameIdEncrypted: false
                requestedAuthnContext: true
                signMetadata: false
                wantXMLValidation: true
                signatureAlgorithm: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'
                digestAlgorithm: 'http://www.w3.org/2001/04/xmlenc#sha256'
            contactPerson:
                technical:
                    givenName: 'Kimai Admin'
                    emailAddress: 'Admin@example.com'
                support:
                    givenName: 'Kimai Support'
                    emailAddress: 'Kimai @example.com'
            organization:
                en:
                    name: 'Kimai'
                    displayname: 'Kimai'
                    url: 'https://Kimai .example.com'

Today, I didn't change anything except for changing session.gc_maxlifetime = 10800, why does this error occur?

How can I resolve that issue?

**Comment by seleondevel** [comment]
Google SAML login works well, but I don't know why this error sometimes happens.

I've logged out several times now and I'm trying to log in, and this error is happening intermittently.

**Reply by kevinpapst** [reply]
Using Google SAML since years: never seen before.

Probably some network device/proxy caching SAML handshakes/responses.

**Reply by seleondevel** [reply]
I don't have any server network firewall settings, but I don't know why I keep getting intermittent errors.
Regarding SAML, I haven't modified the code separately, but I don't know the cause of the error.
